helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixManager.java

```diff
@@ -1223,7 +1225,13 @@ boolean isFlapping() {
   public void handleStateChanged(KeeperState state) {
     switch (state) {
     case SyncConnected:
+      // If manager is disconnected and is the leader controller, then reset and init leaderElectionHandler. This
+      // will force the controller to refresh and pick up any events that were missed during disconnect.
+      if (_managerDisconnectedPastTimeout && _controller != null && isLeader()) {
+        _controller.scheduleOnDemandRebalance(0, true);
+      }
```

Comment:
This is not a proper handling. If you need the full refresh, you can clean up the cache and let pipeline to read data. This will trigger another full refresh after pipeline running.